src/completers/git.zsh

```diff
@@ -689,6 +828,38 @@ _fzf_complete_git-refs_post() {
     done
 }
 
+_fzf_complete_git-notes() {
+    local fzf_options=$1
+    shift
+
+    _fzf_complete --ansi --tiebreak=index ${(Q)${(Z+n+)fzf_options}} -- $@$prefix_option$prefix_ref < <(
+        git for-each-ref refs/notes --format='%(refname:short) %(contents:subject)' 2> /dev/null |
+            _fzf_complete_tabularize ${fg[yellow]}
+    )
+}
+
+_fzf_complete_git-notes_post() {
+    awk '{ print $1 }'
+}
+
+_fzf_complete_git-show-files() {
+    local fzf_options=$1
+    shift
+
+    _fzf_complete --ansi --read0 --print0 ${(Q)${(Z+n+)fzf_options}} -- $@$prefix_option < <(
+        git show --pretty=tformat: --name-only -z ${(ps: :)treeish} 2> /dev/null | sort -u -z
```

Comment:
How about using built-in functionarity in Zsh for deduplication? Probably it should make this faster and more platform-compatible.

Reply:
Working example looks like:
```zsh
local result=$(git show --pretty=format: --name-only -z ${(ps: :)treeish} 2> /dev/null)
echo -n ${(pj:\0:)${(u)${(0)result}}}
```